ads/README.md

```diff
@@ -0,0 +1,94 @@
+# Integrating ad networks into AMP
+
+See also our [ad integration guidelines](../3p/README.md#ads).
+
+## Overview
+Ads are just another external resource and must play within the same constraints placed on all resources in AMP. We aim to support a large subset of existing ads with little or no changes to how the integrations work. Our long term goal is to further improve the impact of ads on the user experience through changes across the entire vertical client side stack.
+
+
+## Constraints
+A summary of constraints placed on external resources such as ads in AMP HTML:
+
+- Because AMPs are served on HTTPS and ads cannot be proxied ads must be served on HTTPS.
+- The size of an ad unit must be static. It must be knowable without fetching the ad and it cannot change at runtime except through https://github.com/ampproject/amphtml/issues/728.
+- If placing the ad requires running JavaScript (assumed to be true for 100% of ads served through networks), the ad must be placed on an origin different from the AMP document itself.
+Reasons include:
+  - Improved security.
+  - Takes synchronous HTTP requests made by the ad out of the critical path of the primary page.
+  - Allows browsers to run the ad in a different process from the primary page (even better security and prevents JS inside the ad to block the main page UI thread).
+  - Prevents ads doing less than optimal things to measure user behavior and other interference with the primary page.
+- The AMP runtime may at any moment decide that there are too many iframes on a page and that memory is low. In that case it would unload ads that were previously loaded and are no longer visible. It may later load new ads in the same slot if the user scrolls them back into view.
+
+
+## The iframe sandbox
+
+The ad itself is hosted within a document that has an origin different from the primary page.
+
+### Information available to the ad
+We will provide the following information to the ad:
+
+- window.context.referrer contains the origin of the referrer value of the primary document if available.
+- document.referrer will typically contain the URL of the primary document. This may change in the future (See next value for a more reliable method).
+- window.context.location.href contains the sanitized location.href value of the primary document.
+- window.context.canonicalUrl contains the canonical URL of the primary document as defined by its link=canonical tag.
+- ad viewability: Tracked in https://github.com/ampproject/amphtml/issues/720
+- window.context.noContentAvailable is a function that the ad system can call if the ad slot was not filled. The container page will then react by showing placeholder content (not by collapsing the ad slot).
```

Comment:
Maybe mention that "iframe resizing" rules will apply or "placeholder" content will be shown. E.g. we can collapse space if outside of the viewport.